lnwallet/rpcwallet/rpcwallet.go

```diff
@@ -651,109 +525,162 @@ func (r *RPCKeyRing) SignOutputRaw(tx *wire.MsgTx,
+		signDesc.Output,
+	)
 	if err != nil {
 		return nil, err
 	}
+	signDesc.WitnessScript = witnessProgram
 
-	resp, err := r.signerClient.ComputeInputScript(ctxt, rpcSignReq)
+	// Let's give the TX to the remote instance now, so it can sign the
+	// input.
+	sig, err := r.remoteSign(tx, signDesc, sigScript)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error signing with remote instance: %v",
+			err)
 	}
 
+	// ComputeInputScript currently is only used for P2WKH and NP2WKH
+	// addresses. So the last item on the stack is always the compressed
+	// public key.
 	return &input.Script{
-		Witness:   resp.InputScripts[0].Witness,
-		SigScript: resp.InputScripts[0].SigScript,
-	}, nil
-}
-
-// coinFromOutPoint attempts to locate details pertaining to a coin based on
-// its outpoint. If the coin isn't under the control of the backing watch-only
-// wallet, then an error is returned.
-func (r *RPCKeyRing) coinFromOutPoint(op wire.OutPoint) (*chanfunding.Coin,
-	error) {
-
-	inputInfo, err := r.WalletController.FetchInputInfo(&op)
-	if err != nil {
-		return nil, err
-	}
-
-	return &chanfunding.Coin{
-		TxOut: wire.TxOut{
-			Value:    int64(inputInfo.Value),
-			PkScript: inputInfo.PkScript,
+		Witness: wire.TxWitness{
+			append(sig.Serialize(), byte(signDesc.HashType)),
+			addr.PubKey().SerializeCompressed(),
 		},
-		OutPoint: inputInfo.OutPoint,
+		SigScript: sigScript,
 	}, nil
 }
 
-// toRPCSignReq converts the given raw transaction and sign descriptors into
-// their corresponding RPC counterparts.
-func toRPCSignReq(tx *wire.MsgTx,
-	signDesc *input.SignDescriptor) (*signrpc.SignReq, error) {
+// remoteSign signs the input specified in signDesc of the given transaction tx
+// using the remote signing instance.
+func (r *RPCKeyRing) remoteSign(tx *wire.MsgTx, signDesc *input.SignDescriptor,
+	sigScript []byte) (input.Signature, error) {
 
-	if signDesc.Output == nil {
-		return nil, fmt.Errorf("need output to sign")
+	packet, err := packetFromTx(tx)
+	if err != nil {
+		return nil, fmt.Errorf("error converting TX into PSBT: %v", err)
 	}
 
-	var buf bytes.Buffer
-	if err := tx.Serialize(&buf); err != nil {
-		return nil, err
+	// Catch incorrect signing input index, just in case.
+	if signDesc.InputIndex < 0 || signDesc.InputIndex >= len(packet.Inputs) {
+		return nil, fmt.Errorf("invalid input index in sign descriptor")
 	}
+	in := &packet.Inputs[signDesc.InputIndex]
+	txIn := tx.TxIn[signDesc.InputIndex]
 
-	rpcSignDesc := &signrpc.SignDescriptor{
-		KeyDesc: &signrpc.KeyDescriptor{
-			KeyLoc: &signrpc.KeyLocator{
-				KeyFamily: int32(signDesc.KeyDesc.Family),
-				KeyIndex:  int32(signDesc.KeyDesc.Index),
+	// Make sure we actually know about the input. We either have been
+	// watching the UTXO on-chain or we have been given all the required
+	// info in the sign descriptor.
+	info, err := r.WalletController.FetchInputInfo(&txIn.PreviousOutPoint)
```

Comment:
Do we need to check the err against `ErrNotMine` to make sure it's saying we don't have it?

Reply:
The `FetchInputInfo` is just used to also get the full previous TX for the `NonWitnessUtxo` field, which might be required by hardware wallets. Maybe you could say this is a premature optimization to make the code more generic.
But in most cases we populate what is required in the sign descriptor and don't actually need the full input info. So we don't really care what error is returned as we have a fallback in most cases anyway.